File: recursion_js/3_decimal_to_binary.js
function decimalToBinary(decimal, result = "") {
    if (decimal === 0) {
        return result || "0"; // handle input 0
    }
    result = (decimal % 2) + result;
    return decimalToBinary(Math.floor(decimal / 2), result);
}

console.log(decimalToBinary(0));  // "0"
console.log(decimalToBinary(1));  // "1"
console.log(decimalToBinary(2));  // "10"
console.log(decimalToBinary(5));  // "101"
console.log(decimalToBinary(10)); // "1010"
console.log(decimalToBinary(19)); // "10011"

File: navcodes/8_recursion/3_decimal_to_binary.py
def decimal_to_binary(decimal, result=""):
    if decimal == 0:
        return result or "0"
    
    result = str(decimal%2) + result
    return decimal_to_binary(decimal//2, result)

if __name__ == "__main__":
    print(decimal_to_binary(0))
    print(decimal_to_binary(1))
    print(decimal_to_binary(2))
    print(decimal_to_binary(5))
    print(decimal_to_binary(10))
    print(decimal_to_binary(19))